docs/development/awk/intro-to-awk/intro-to-awk.md

```diff
@@ -0,0 +1,732 @@
+    but it also matches any sequence of spaces and tabs. Additionally, any number
+	of leading or trailing spaces and tabs is being ignored. If the value of `FS` becomes the null
+	string, then each character in the current line becomes a separate field.
+- The `FILENAME` built-in variable holds the filename of the file currently being
+    processed. Notice that you cannot use `FILENAME` inside the `BEGIN` block because
+	it is not defined yet.
+- The `NR` variable is used for keeping track of the total number of records that have been read so far.
+- The `FNR` variable is used for keeping the total number of records that have been read from the current input file only.
+- The `IGNORECASE` variable is used for telling AWK to ignore case in all of its comparisons or regular expressions.
+- The `ARGC` and `ARGC` variables hold the command line arguments of an AWK program.
+
+### User Defined Variables in AWK
+
+The way you use an AWK variable the first time will define its type. The following
+AWK code uses the `n` variable to count the number of lines that contain a given
+string, which is this case is the word `three`:
+
+    awk '/three/ { n++ }; END { print n+0 }'
+
+The above code says AWK to execute the `n++` code each time there is a match to the
+`/three/` regular expression. When the processing is done, the code in `END` will be
+executed, which just prints the current value of `n` converted to number.
+
+If you provide some input to it, either a filename or user input from the console, you
+will get some output from the AWK code.
+
+## AWK Basics
+
+In this section of the guide you will learn more about the basics of AWK. AWK has support
+for **arrays** and **associative arrays**. It also supports functions and user
+defined functions. AWK programs can have various types of blocks (*patterns*) like
+`BEGIN` and `END` that are executed at the time that is specified by AWK.
+
+### Arrays and Associative Arrays in AWK
+
+This part of the guide will introduce you to AWK **Arrays** and **Associative Arrays**.
+At this point it would be good to know that arrays and associative arrays in AWK are
+exactly the same thing. What logically distinguishes an array from an associative array
+is that the indices of an array are positive integers whereas the keys of an associative
+array are strings. However, AWK uses string keys internally in both cases! Additionally, you
+should use the `delete` statement for removing an element from an array.
+
+Notice that associative arrays are called *hashes* in other programming languages.
+
+#### An example
+
+The following example illustrates associative arrays in AWK using the code found in `arrays.awk`.
+
+{{< file "arrays.awk" awk >}}
+BEGIN {
+	a[0] = 1;
+	a[1] = 2;
+	a[2] = 3;
+	a[3] = 4;
+
+	for (i in a)
+		print "Index:", i, "with value:", a[i];
+
+	print "Adding two elements and deleting a[0]";
+
+	a["One"] = "OneOne";
```

Comment:
The "OneOne" and "TwoTwo" examples are a little confusing--is there something more legible that we can use?

Reply:
Used `Two_value` and `One_value`.